### PR TITLE
Compiler rt

### PR DESCRIPTION
**Pull request by draganmladjenovic**
Checked with:

make  -S compiler-rt -B build_compiler_rt  -DCMAKE_C_COMPILER=nanomips-elf-gcc  -DCMAKE_C_FLAGS="" -DLLVM_CONFIG_PATH=./build/bin/llvm-config -DCOMPILER_RT_INCLUDE_TESTS=ON \
-DCOMPILER_RT_BUILTINS_ENABLE_PIC=OFF -DCMAKE_C_COMPILER_TARGET="nanomips-elf" -DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON -DCOMPILER_RT_BAREMETAL_BUILD=ON -DCOMPILER_RT_EMULATOR="nanomips-elf-run" \
-DBUILTINS_TEST_TARGET_CFLAGS="-Tmti32.ld"
cd build_compiler_rt
make check-compiler-rt
Failed Tests (4):
  Builtins-nanomips-linux :: compiler_rt_scalbnf_test.c
  Builtins-nanomips-linux :: trampoline_setup_test.c
  Builtins-nanomips-linux :: truncdfhf2_test.c
  Builtins-nanomips-linux :: truncsfhf2_test.c

Testing Time: 17.72s
  Unsupported: 106
  Passed     : 104
  Failed     :   4

make  -S compiler-rt -B build_compiler_rt  -DCMAKE_C_COMPILER=./build/bin/clang  -DCMAKE_C_FLAGS="--gcc-toolchain=<nanomips-elf-gcc-path>" \
-DLLVM_CONFIG_PATH=./build/bin/llvm-config -DCOMPILER_RT_INCLUDE_TESTS=ON -DCOMPILER_RT_BUILTINS_ENABLE_PIC=OFF -DCMAKE_C_COMPILER_TARGET="nanomips-elf" -DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON \
-DCOMPILER_RT_BAREMETAL_BUILD=ON -DCOMPILER_RT_EMULATOR="nanomips-elf-run" -DBUILTINS_TEST_TARGET_CFLAGS="--target=nanomips-elf --gcc-toolchain=<nanomips-elf-gcc-path> -Tmti32.ld"
cd build_compiler_rt
make check-compiler-rt
Failed Tests (1):
  Builtins-nanomips-linux :: compiler_rt_scalbnf_test.c

Testing Time: 15.10s
  Unsupported: 106
  Passed     : 107
  Failed     :   1

Resulting binary named libclang_rt.builtins-nanomips.a

### CODE REVIEW

**Comment by djtodoro** [comment]
@draganmladjenovic did we double check if the `compiler-rt` is built with optimizations enabled?

**Review by djtodoro — APPROVED** [review]

**Comment by farazs-github** [comment]
@djtodoro  As far as I can make out, no optimizations are enabled by default.
The default options for gcc are: -Os -mno-gpopt -ffunction-sections -fdata-sections
I will add these to the build script.

File: clang/lib/Driver/ToolChains/Gnu.cpp
Comment (farazs-github) [inline clang/lib/Driver/ToolChains/Gnu.cpp:1]: I think this commit should be split out from this pull request . It serves no purpose when building compiler-rt using gcc.

**Review by farazs-github — APPROVED** [review]
LGTM

**Pull request merged**